File: 251002/Samoilenko/publisher/src/main/java/com/example/discussion/service/mapper/WriterMapper.java
package com.example.discussion.service.mapper;

import com.example.discussion.dto.WriterRequestTo;
import com.example.discussion.dto.WriterResponseTo;
import com.example.discussion.model.Writer;
import org.mapstruct.Mapper;
import org.mapstruct.Mapping;
import org.mapstruct.MappingTarget;

@Mapper(componentModel = "spring")
public interface WriterMapper {
    @Mapping(source = "firstname", target = "firstname")
    @Mapping(source = "lastname", target = "lastname")
    @Mapping(target = "id", ignore = true)  // Игнорируем ID при создании
    Writer toEntity(WriterRequestTo dto);

    @Mapping(source = "firstname", target = "firstname")
    @Mapping(source = "lastname", target = "lastname")
    WriterResponseTo toDto(Writer writer);

    @Mapping(target = "id", ignore = true)  // Игнорируем ID при обновлении
    void updateEntityFromDto(WriterRequestTo dto, @MappingTarget Writer entity);
}

File: 251003/Pankratiev/rest/src/main/kotlin/bsuir/dc/rest/mapper/WriterMapper.kt
package bsuir.dc.rest.mapper

import bsuir.dc.rest.dto.from.WriterRequestTo
import bsuir.dc.rest.dto.to.WriterResponseTo
import bsuir.dc.rest.entity.Writer

fun WriterRequestTo.toEntity(): Writer = Writer(
    id = id,
    login = this.login,
    password = this.password,
    firstname = this.firstname,
    lastname = this.lastname
)

fun Writer.toResponse(): WriterResponseTo = WriterResponseTo(
    id = this.id,
    login = this.login,
    firstname = this.firstname,
    lastname = this.lastname
)
